common/src/main/java/me/shedaniel/architectury/event/events/InteractionEvent.java

```diff
@@ -34,46 +34,132 @@
+        /**
+         * Invoked whenever a player uses an item on a block.
+         * Equal to the Forge {@code PlayerInteractEvent.RightClickItem} event and Fabric's {@code UseItemCallback}.
+         *
+         * @param player The player right clicking the block.
+         * @param hand The hand that is used. You can get the {@link ItemStack} with {@link Player#getItemInHand(InteractionHand)}
+         * @return Whenever the return is not {@link InteractionResult#PASS}, the result value is used and the event is canceled.
+         */
         InteractionResultHolder<ItemStack> click(Player player, InteractionHand hand);
     }
     
     interface ClientRightClickAir {
+        /**
+         * Invoked whenever a player right clicks the air.
+         * This only occurs on the client.
+         * Equal to the Forge {@code PlayerInteractEvent.RightClickEmpty} event.
+         * 
+         * @param player The player. Always {@link net.minecraft.client.player.LocalPlayer}
+         * @param hand The hand used.
+         */
         void click(Player player, InteractionHand hand);
     }
     
     interface ClientLeftClickAir {
+        /**
+         * Invoked whenever a player left clicks the air.
+         * This only occurs on the client.
+         * Equal to the Forge {@code PlayerInteractEvent.LeftClickEmpty} event.
+         *
+         * @param player The player. Always {@link net.minecraft.client.player.LocalPlayer}
+         * @param hand The hand used.
+         */
         void click(Player player, InteractionHand hand);
     }
     
     interface InteractEntity {
+        /**
+         * Invoked whenever a player right clicks an entity.
+         * Equal to the Forge {@code PlayerInteractEvent.EntityInteract} event.
+         * 
+         * @param player The player clicking the entity.
+         * @param entity Then entity the player clicks.
+         * @param hand The used hand.
+         * @return If the return value is not {@link InteractionResult#PASS}, event is cancelled and the used result is passed as return value.
+         */
         InteractionResult interact(Player player, Entity entity, InteractionHand hand);
     }
     
+    /**
+     * @deprecated use {@link BlockEvent#BREAK}
+     */
+    @Deprecated
     interface BlockBreak {
         InteractionResult breakBlock(Player player, BlockPos pos, BlockState state);
     }
     
     interface FarmlandTrample {
+        /**
+         * Called when an entity attempts to trample farmland.
+         * Equal to the Forge {@code BlockEvent.FarmlandTrampleEvent} event.
+         * 
+         * @param world The level where the block and the player are located in.
+         * @param pos The position of the block.
+         * @param state The state of the block.
+         * @param distance The distance of the player to the block.
+         * @param entity The entity trampling.
+         * @return If any value is returned, the trampling is canceled.
```

Comment:
if any value? I would just do the following:
```suggestion
         * @return The event callback result.
```